pkg/config/config.go

```diff
@@ -463,6 +463,8 @@ func (p Package) FullCopyright() string {
 type Needs struct {
 	// A list of packages needed by this pipeline
 	Packages []string
+	// A list of environment variables to inject to the build or test config.
+	Environment map[string]string
```

Comment:
I'm not really convinced by this API as `needs.environment` can be confusing compared to `environment`.